### PR TITLE
feat[SWA-291]: zip code field

### PR DESCRIPTION
**Pull request by patrinoua**
https://user-images.githubusercontent.com/36475005/162452374-aebd680e-4cc0-4c39-b2c7-8cdf9ac64d6f.mov

This PR resolves [SWA-291]

### PR Checklist (tick all before merging)

- [ ] Feature flag around my changes.
- [ ] App state migration.
- [ ] Screenshots/videos of my changes.
- [ ] Tested on **iOS** and **Android**.
- [ ] Tests/stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[SWA-291]: https://artsyproduct.atlassian.net/browse/SWA-291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/ArtworkDetailsForm.tsx [hunk 1]
```diff
@@ -204,6 +204,18 @@ export const ArtworkDetailsForm: React.FC = () => {
...
+      {countriesRequirePostalCode.includes(values.location.countryCode.toUpperCase()) && (
+        <>
+          <Spacer m={2} />
+          <Input
+            title="Zip/Postal code"
+            placeholder="Zip/Postal Code Where Artwork Is Located"
+            testID="Submission_ZipInput"
+            value={values.zipCode}
+            onChangeText={(e) => setFieldValue("zipCode", e)}
+          />
```
Comment (kiryl-zubarau) [inline src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/ArtworkDetailsForm.tsx:216]: Do we show the error message for the wrong postal code?

File: src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts [hunk 2]
```diff
@@ -53,15 +55,50 @@ export const artworkDetailsEmptyInitialValues: ArtworkDetailsFormModel = {
...
+  "LV", // Latvia
+  "MC", // Monaco
+  "NO", // Norway
+  "PL", // Poland
+  "PT", // Portugal
+  "ES", // Spain
+  "SE", // Sweden
+  "CH", // Switzerland
+  "GB", // United Kingdom
+  // "NL", // Netherlands
```
Comment (AlicanAkyuz) [inline src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts:97]: NL commented out here 👀

**Comment by AlicanAkyuz** [comment]
@patrinoua could you please add a screenshot/video of how the input looks and works, with success and error for US and a non-US country? thank you!

File: src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/ArtworkDetailsForm.tsx [hunk 3]
```diff
@@ -204,6 +206,21 @@ export const ArtworkDetailsForm: React.FC = () => {
...
+        <>
+          <Spacer m={2} />
+          <Input
+            title="Zip/Postal code"
+            placeholder="Zip/Postal Code Where Artwork Is Located"
+            testID="Submission_ZipInput"
+            value={values.zipCode}
+            onBlur={() => setIsZipInputFocused(false)}
+            onFocus={() => setIsZipInputFocused(true)}
+            error={!isZipInputFocused && values.zipCode && errors.zipCode ? errors.zipCode : ""}
```
Comment (kiryl-zubarau) [inline src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/ArtworkDetailsForm.tsx:219]: @patrinoua we should also show `Please enter a valid zip/postal code for your region` message for other countries if the field empty.

File: src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts [hunk 4]
```diff
@@ -53,15 +55,50 @@ export const artworkDetailsEmptyInitialValues: ArtworkDetailsFormModel = {
   utmMedium: "",
   utmSource: "",
   utmTerm: "",
+  zipCode: "",
```
Comment (MounirDhahri) [inline src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts:58]: **thought:** I would have assumed that the `zipCode` would be part of the location and not its own separate field, along with the `countryCode`.

**Review by AlicanAkyuz — APPROVED** [review]

File: src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts (hunk 4 above)
Comment (kiryl-zubarau) [reply src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts]: Updated

**Review by iskounen — APPROVED** [review]

**Review by AlicanAkyuz — APPROVED** [review]

File: src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts (hunk 4 above)
Comment (MounirDhahri) [reply src/app/Scenes/Consignments/Screens/SubmitArtworkOverview/ArtworkDetails/validation.ts]: nice! thanks

**Comment by patrinoua** [comment]
@kiryl-zubarau Thank you for finishing this up 🙏

**Pull request merged**